lib/std/special/compiler_rt/clear_cache.zig

```diff
@@ -45,28 +49,31 @@ pub fn clear_cache(start: usize, end: usize) callconv(.C) void {
         @compileError("TODO");
         // FlushInstructionCache(GetCurrentProcess(), start, end - start);
     } else if (arm32 and !apple) {
-        if (os == .freebsd or os == .netbsd) {
-            //  struct arm_sync_icache_args arg;
-            //
-            //  arg.addr = (uintptr_t)start;
-            //  arg.len = (uintptr_t)end - (uintptr_t)start;
-            //
-            //  sysarch(ARM_SYNC_ICACHE, &arg);
-            @compileError("TODO: implement for NetBSD/FreeBSD");
-        } else if (os == .linux) {
-            const result = std.os.linux.syscall3(std.os.linux.SYS_cacheflush, start, end, 0);
-            std.debug.assert(result == 0);
-        } else {
-            @compileError("no __clear_cache implementation available for this target");
+        switch (os) {
+            .freebsd, .netbsd => {
+                var arg = arm_sync_icache_args{
+                    .addr = start,
+                    .len = end - start,
+                };
+                const result = sysarch(ARM_SYNC_ICACHE, @ptrToInt(&arg));
+                std.debug.assert(result == 0);
+            },
+            .linux => {
+                const result = std.os.linux.syscall3(std.os.linux.SYS_cacheflush, start, end, 0);
+                std.debug.assert(result == 0);
+            },
+            else => @compileError("TODO"),
         }
     } else if (os == .linux and mips) {
-        @compileError("TODO");
-        //const uintptr_t start_int = (uintptr_t)start;
-        //const uintptr_t end_int = (uintptr_t)end;
-        //syscall(__NR_cacheflush, start, (end_int - start_int), BCACHE);
+        const flags = 3; // ICACHE | DCACHE
```

Comment:
Factor out the magic numbers?

Also why is this not `BCACHE` as it was in the removed comment?

Reply:
Because this syscall is a joke
> This function always behaves as if BCACHE has been passed for the cache
argument and does not do any error checking on the cache argument.

I've only left the flags there to warn the user that both the caches are flushed.

> Also why is this not BCACHE as it was in the removed comment?

Because `BCACHE` is `ICACHE | DCACHE` and the `B` is meaningless to me (was it for Both? Who knows)